imperial_coldfront_plugin/gid.py

```diff
@@ -0,0 +1,40 @@
+"""Functionality to get a new GID value."""
+
+from coldfront.core.allocation.models import Allocation
+from django.conf import settings
+from django.db.models import Max
+
+
+def get_new_gid() -> int:
+    """Get a new GID value.
+
+    This function checks the existing GID values in the database and returns
+    the next available GID within the specified ranges. If no GID is
+    available, it raises a ValueError.
+
+    Returns:
+        int: The next available GID value.
+
+    Raises:
+        ValueError: If no available GID is found in the configured ranges.
+    """
+    existing_gids = Allocation.objects.filter(
+        allocationattribute__allocation_attribute_type__name="GID"
+    ).values_list("allocationattribute__value", flat=True)
+
+    # Get the maximum GID value already assigned
+    max_gid = existing_gids.aggregate(Max("allocationattribute__value"))[
+        "allocationattribute__value__max"
+    ]
```

Comment:
This looks like it should work but it doesn't because of the funny way they handled how attribute types work where all values are actually internally stored as strings. This is a bit crude but should do the trick (assuming the attribute type is changed to `Int`).
 ```suggestion
    existing_gids = AllocationAttribute.objects.filter(
        allocation_attribute_type__name="GID"
    )

    # Get the maximum GID value already assigned
    max_gid = max(eg.typed_value() for eg in existing_gids)
```